### PR TITLE
Add `@FuzzTest#maxExecutions` and improve `LifecycleFuzzTest`

### PR DESCRIPTION
**Pull request by fmeum**

### CODE REVIEW

File: src/main/java/com/code_intelligence/jazzer/junit/FuzzTest.java [hunk 1]
```diff
@@ -103,9 +103,22 @@
...
+  /**
+   * If set to a positive number, the fuzz test function will be executed at most this many times
+   * during fuzzing.
+   *
+   * <p>Prefer this element over {@link #maxDuration()} if you want to ensure comparable levels of
+   * fuzzing across machine's with different performance characteristics.
+   *
+   * <p>This option has no effect during regression testing.
+   */
+  long maxExecutions() default 0;
```
Comment (fmeum) [inline src/main/java/com/code_intelligence/jazzer/junit/FuzzTest.java:121]: Happy to bikeshed this name. I found "runs" to clash with our concept of "fuzzing runs" while "executions" goes well with "exec/s" metrics in libFuzzer output.

Comment (bertschneider) [inline src/main/java/com/code_intelligence/jazzer/junit/FuzzTest.java:121]: We normally set `-runs=0` to only execute corpus entries. Should the annotation have  a different semantic and couldn't this  be confusing? I'm fine either way, though, but would add `0` as infinite runs to the description.

File: src/main/java/com/code_intelligence/jazzer/junit/AgentConfiguringArgumentsProvider.java [hunk 2]
```diff
@@ -37,7 +37,8 @@ public Stream<? extends Arguments> provideArguments(ExtensionContext extensionCo
     // FIXME(fmeum): Calling this here feels like a hack. There should be a lifecycle hook that runs
     //  before the argument discovery for a ParameterizedTest is kicked off, but I haven't found
     //  one.
-    FuzzTestExecutor.configureAndInstallAgent(extensionContext, fuzzTest.maxDuration());
+    FuzzTestExecutor.configureAndInstallAgent(
```
Comment (bertschneider) [inline src/main/java/com/code_intelligence/jazzer/junit/AgentConfiguringArgumentsProvider.java:40]: Could you please add a comment why the agent needs to be installed here and in `FuzzTestExtensions`? We discussed this, but I forgot again 😬

File: src/main/java/com/code_intelligence/jazzer/junit/FuzzTest.java (hunk 1 above)
Comment (fmeum) [reply src/main/java/com/code_intelligence/jazzer/junit/FuzzTest.java:121]: We do have the regression test mode for this, so I don't think `-runs=0` has much use with JUnit-based fuzz tests. But it should still be supported (e.g. for use in cifuzz) as command-line arguments take precedence over this value.

I added an explanation of the default case to the docs.

File: examples/junit/src/test/java/com/example/LifecycleFuzzTest.java [hunk 3]
```diff
@@ -16,72 +16,125 @@
...
+      // TODO: Fuzz tests currently don't run before each and after each methods between fuzz test
+      // invocations.
+      expectedEvents.addAll(Collections.nCopies(3, "lifecycleFuzz"));
+      expectedEvents.addAll(expectedAfterEachEvents);
+    }
+
+    expectedEvents.add("afterAll");
+
+    assertThat(events).containsExactlyElementsIn(expectedEvents).inOrder();
     throw new IOException();
```
[the remark this answers is not in the document]
Comment (fmeum) [reply examples/junit/src/test/java/com/example/LifecycleFuzzTest.java]: I am doing that now and also added a comment explaining why this isn't fully straightforward.

File: src/main/java/com/code_intelligence/jazzer/junit/AgentConfiguringArgumentsProvider.java (hunk 2 above)
Comment (fmeum) [reply src/main/java/com/code_intelligence/jazzer/junit/AgentConfiguringArgumentsProvider.java:43]: Good point, done :-)

**Review by bertschneider — APPROVED** [review]
Nice addition!

**Pull request merged**